docs/build/build-options.md

```diff
@@ -421,6 +422,55 @@ to use either of those (`ENABLE_STDCXX_SYNC` excludes `ENABLE_MONOTONIC_CLOCK`).
 When ON, this option enables the standard C++ `thread` and `chrono` libraries 
 (available since C++11) to be used by SRT instead of the `pthreads` libraries.
 
+#### ENABLE_PKTINFO
+**`--enable-pktinfo`** (default: OFF)
+
+This allows the use of `IP_PKTINFO` control message which can be used to
+extract the true target IP address from the incoming UDP packets to a listener
+bound to a so-called "any" address (`0.0.0.0` for IPv4, AKA `INADDR_ANY`, or
+`::` for IPv6, AKA `in6addr_any`, or simply when the application clears the
+`sockaddr*` structure and only sets the port number), and then set this address
+as source IP address when sending packets to the peer. This solves the problem,
+when the routing rules in the agent's host make that sending a packet to a
+given IP address would use a different source IP address than the one that was
+set as a target IP address in the UDP packet from the peer. The peer will
+reject such a packet as a suspected MITM attempt, and this will lead to a
+connection failure.
+
+This problem has been observed in the case, when the agent's host has at least
+two IP addresses that share the same broadcast prefix and it is being contacted
+by the peer using other than the first of these addresses. For example:
+
+The host has set the following local IP addresses:
+
+* 192.168.10.5 - routing to 192.168.10.1
+* 10.10.5.10 - routing to 10.10.5.1
+* 10.10.5.20 - routing to 10.10.5.1
+
+In all of them the netmask is 255.255.255.0, which means that the second
+and third IP address share the same broadcast prefix, which is 10.10.5.0.
+The problematic case is when an agent running on this host will be contacted
+by a peer using the address 10.10.5.20.
+
+In such a case the source address set in the UDP packet sent back to the peer
+will always be the first of these addresses (10.10.5.10), which will be
+different than the one the peer used to contact (10.10.5.20). The peer will
+then rejeect such a packet by having the source address different than the one
+it tries to contact.
+
+This problem occurs only with listener sockets bound to "any" address - when it
+is bound to a specific IP address, this will be always set as source address
+for outgoing packets.
+
+By enabling this feature SRT mitigates this problem by first reading the real
+target IP address from the incoming handshake packet, and then setting this
+very address forcefully to the source address field of every UDP packet sent
+from this socket. This behavior is also consistent with TCP.
+
+Note that this feature is available only on certain platforms. Notably the BSD
+systems are known to not provide this feature and the current implementation
+doesn't support this feature on Windows systems.
```

Comment:
```suggestion
Note that this feature is not supported on BSD or Windows systems.
```

Reply:
```suggestion
Note that this feature is not supported on BSD systems. On Windows systems it is possible, but the API used currently for sending and receiving UDP packets does not allow the use of CMSG messages. Such an API exists, so for Windows this would only require changing the API functions used for sending/receiving (use `WSASendMsg` instead of `WSASendTo` etc.).
```

Reply:
Well, I wouldn't go into such details - maybe this way:

Note that this feature is not supported on BSD systems (not possible due to lacking IP_PKTINFO feature). On Windows it is theoretically possible, but currently not implemented.